aladdin-container.sh

```diff
@@ -91,33 +106,38 @@ function environment_init() {
 
     _handle_aws_config
 
-    # Make sure we are on local or that cluster has been created before initializing helm, creating namespaces, etc
-    if "$IS_LOCAL" || ( kops export kubecfg --name $CLUSTER_NAME > /dev/null && \
-        kops validate cluster --name $CLUSTER_NAME ) > /dev/null; then
-
-        if "$IS_LOCAL"; then
-            mkdir -p $HOME/.kube/
-            cp $HOME/.kube_local/config $HOME/.kube/config
-            # Replace e.g. /Users/ptr/.minikube with /root/.minikube in the minikube conf dir path
-            # (since that's where it will be mounted within the aladdin container)
-            sed 's|: \?.*\.minikube|: /root/.minikube| ; /: \/root\/.minikube/ s|\\|/|g' $HOME/.kube_local/config > $HOME/.kube/config
-        else
+    if "$IS_LOCAL"; then
+        mkdir -p $HOME/.kube/
+        cp $HOME/.kube_local/config $HOME/.kube/config
+        # Replace e.g. /Users/ptr/.minikube with /root/.minikube in the minikube conf dir path
+        # (since that's where it will be mounted within the aladdin container)
+        sed 's|: \?.*\.minikube|: /root/.minikube| ; /: \/root\/.minikube/ s|\\|/|g' $HOME/.kube_local/config > $HOME/.kube/config
+    elif test -z "$(get_or_set_cache "kops_$CLUSTER_NAME")" || ! test -f "$HOME/.kube_local/$CLUSTER_NAME.config"; then
+        echo "Getting cluster config and verifying cluster using kops"
+        if ( kops export kubecfg --name $CLUSTER_NAME > /dev/null && \
+            kops validate cluster --name $CLUSTER_NAME ) > /dev/null; then
```

Comment:
previously we were doing these kops commands on each command that wasn't on the minikube cluster